app/src/main/java/com/android/unio/ui/event/EventCard.kt

```diff
@@ -33,118 +34,83 @@ import androidx.compose.ui.draw.clip
 import coil.request.ImageRequest
 import com.android.unio.R
-import com.android.unio.model.association.Association
 import com.android.unio.model.event.Event
 import com.android.unio.model.event.EventType
-import com.android.unio.model.event.EventViewModel
-import com.android.unio.model.event.MockEventRepository
-import com.android.unio.model.event.PreviewEventViewModel
-import com.android.unio.model.firestore.MockReferenceList
-import com.android.unio.model.map.Location
-import com.android.unio.model.user.MockUserRepository
+import com.android.unio.model.user.UserViewModel
+import com.android.unio.ui.theme.AppTypography
 import com.android.unio.ui.theme.primaryContainerLight
+import com.android.unio.ui.theme.primaryDark
 import com.android.unio.ui.theme.secondaryDark
 import com.android.unio.utils.EventUtils.addAlphaToColor
 import com.android.unio.utils.EventUtils.formatTimestamp
-import com.google.firebase.Timestamp
 import java.text.SimpleDateFormat
-import java.util.Date
 import java.util.Locale
 
-@Preview(showBackground = true)
 @Composable
-fun EventCardPreview() {
-  val sampleEvent =
-      Event(
-          title = "Sample Event",
-          image = "", // Empty string to test the placeholder
-          organisers = MockReferenceList<Association>(),
-          taggedAssociations = MockReferenceList<Association>(),
-          location = Location(0.0, 0.0, "aaaa"),
-          date = Timestamp(Date(2024 - 1900, 6, 20)),
-          catchyDescription = "This is a catchy description.",
-          types = listOf(EventType.TRIP))
-
-  val previewViewModel = PreviewEventViewModel(MockEventRepository(), MockUserRepository())
-  EventCard(event = sampleEvent, viewModel = previewViewModel)
-}
-
-@Composable
-fun EventCard(event: Event, viewModel: EventViewModel) {
+fun EventCard(event: Event, userViewModel: UserViewModel) {
 
   var isSaved by remember { mutableStateOf(false) }
+  val user by userViewModel.user.collectAsState()
 
-  LaunchedEffect(event.uid) {
-    viewModel.isEventSavedForCurrentUser(event.uid) { saved -> isSaved = saved }
-  }
+  LaunchedEffect(event.uid) { isSaved = userViewModel.isEventSavedForCurrentUser(event.uid) }
 
   Column(
       modifier =
           Modifier.fillMaxWidth()
               .padding(vertical = 8.dp)
-              .clickable(onClick = {})
+              // implement clickable feature
               .testTag("event_EventListItem")
               .clip(RoundedCornerShape(10.dp))
               .background(secondaryDark)) {
+
+        // Event image section, displays the main event image or a placeholder if the URL is invalid
+
         Box(modifier = Modifier.fillMaxWidth().height(100.dp)) {
-          // Fallback to a local placeholder image in case of an invalid URI
+          // fallback to a local placeholder image in case of an invalid URI
           val imageUrl = event.image.takeIf { it.isNotEmpty() }?.toUri()
 
-          if (LocalInspectionMode.current) { // preview mode
```

Comment:
Thanks for removing the preview logic, as it does not have its place in the app functionality